api.proto

```diff
@@ -3,13 +3,34 @@ package proto;
 
 option java_package = "io.liftbridge.proto";
 
+
+message RetentionMaxBytes {
+    int64 value = 1; 
+}
+
+message RetentionMaxMessages {
+    int64 value = 1; 
+}
+
+message CompactEnabled {
+    bool value = 1; 
+}
+
 // CreateStreamRequest is sent to create a new stream.
 message CreateStreamRequest {
-    string subject           = 1; // Stream NATS subject
-    string name              = 2; // Stream name (unique per subject)
-    string group             = 3; // Partitions NATS subject amongst group members
-    int32  replicationFactor = 4; // Number of stream replicas
-    int32  partitions        = 5; // Number of stream partitions
+    string                  subject              = 1; // Stream NATS subject
+    string                  name                 = 2;  // Stream name (unique per subject)
+    string                  group                = 3;  // Partitions NATS subject amongst group members
+    int32                   replicationFactor    = 4;  // Number of stream replicas
+    int32                   partitions           = 5;  // Number of stream partitions
+    RetentionMaxBytes       RetentionMaxBytes    = 6;  // The maximum size a stream's log can grow to, in bytes.
+    RetentionMaxMessages    RetentionMaxMessages = 7;  // The maximum size a stream's log can grow to, in messages
+    int64                   RetentionMaxAge      = 8;  // The TTL for stream log segment files 
```

Comment:
How come this was left as a primitive `int64` type? `RetentionMaxAge` has a server default of 1 week. A value of 0 indicates no limit, so seems like we need to use the same pattern as above here?

Reply:
hmm. my bad. I should have gone through all of these configurations to figure out which one need to be wrapped. I.e: the one that uses 0 as no limit